### DISCUSSION TITLE
HTML body parse?

### DISCUSSION THREAD

**Discussion by rgraham7503** [hyperjumptech/monika#842] [General]
Hi I have a messy HTML body response from a page I want to monitor. Its not json. I can get what I need by using curl and grep and cut etc. Is there a way to parse non json body responses and alert if not a match? like parse out component-id and the status and if the status is NOT "operational" then alert? or can we do regex? 

Example: 
`<html>`
`<head>`
  `<title>bla</title>`
`</head>`
`<body>`
`<p>bunch of content</p>`
`<div data-component-id="xxxxxxx"
     class="component-inner-container status-green "
     data-component-status="operational"
     data-js-hook="">`
`<span class="name">`
     ` test alert`
  ` </span>`
`</div>`
`</body>`
`</html>`

**Comment by nicnocquee** [comment]
You can define an alert's query which uses regex: https://monika.hyperjump.tech/guides/alerts

```
x ~= y
```

There was also a discussion about this here https://github.com/hyperjumptech/monika/discussions/649#discussioncomment-2589470

I think we need to put more example how to use regex in the alert.

**Reply by rgraham7503** [reply]
Oh geeze....I'm sorry, I did look before posting but didn't see that discussion. Thats perfect!!! Thank you!

**Reply by rgraham7503** [reply]
I did try a bunch of variations of regex...can you test and make sure its not a bug? getting this error...
ERROR: Lexical error on line 1. Unrecognized text....us.description") ~= "(^.*?\bzt7xx7k572z0

![image](https://user-images.githubusercontent.com/46932722/182278811-e3dec47e-86c1-443d-97ea-4dd1823ab434.png)

**Reply by rgraham7503** [reply]
Ok I was able to get a simple regex to work without encapsulating in ().
